std-bits/aws/src/main/java/org/enso/aws/ProfileReader.java

```diff
@@ -5,15 +5,17 @@
 public class ProfileReader {
   public static final ProfileReader INSTANCE = new ProfileReader();
 
-  private final String[] profiles;
+  private String[] profiles = null;
 
-  private ProfileReader() {
-    var provider = ProfileFileSupplier.defaultSupplier();
-    var profileFile = provider.get();
-    profiles = profileFile.profiles().keySet().toArray(new String[0]);
-  }
+  private ProfileReader() {}
 
   public String[] getProfiles() {
+    if (profiles == null) {
+      var provider = ProfileFileSupplier.defaultSupplier();
+      var profileFile = provider.get();
+      profiles = profileFile.profiles().keySet().toArray(new String[0]);
+    }
```

Comment:
I'm going to be annoying, because it's exactly what I previously suggested.

But after even further though, shouldn't we reload the profiles on every call? Or have some way to reload them? What if someone adds a profile while Enso is running? Currently it seems that they would have to restart it for the changes to take effect, are we fine with that? I guess this may be OK, just asking to be sure.

Reply:
I'll see if I can do anything in the next S3 work. 

Generally a minor thing but if I can detect a change using AWS API will do.